lib/inventory/inventory.go

```diff
@@ -313,6 +300,17 @@ func (h *downstreamHandle) Close() error {
 	return nil
 }
 
+func (h *downstreamHandle) Goodbye(ctx context.Context) error {
```

Comment:
nit: godoc.  More specifically, we probably want the godoc to make a clear explanation of the conceptual difference between sending the goodbye message and actually closing.